### ISSUE TITLE
ipyparallel broken with ipykernel 5.1.0

### ISSUE THREAD

**Issue by basnijholt** [ipython/ipyparallel#363]
The following code hang forever:
```
import ipyparallel
c = ipyparallel.Client('slurm')
res = c[:].execute('import sys')
res.result()
```

These messages were printed in my slurm log-file:
```
2019-04-19 21:23:49.269 [IPEngineApp] ERROR | Exception in message handler:
Traceback (most recent call last):
  File "/gscratch/home/t-banij/miniconda3/envs/py36/lib/python3.6/site-packages/ipykernel/kernelbase.py", line 267, in dispatch_shell
    yield gen.maybe_future(handler(stream, idents, msg))
  File "/gscratch/home/t-banij/miniconda3/envs/py36/lib/python3.6/site-packages/ipyparallel/engine/kernel.py", line 95, in execute_request
    md = self.finish_metadata(parent, md, reply_content)
  File "/gscratch/home/t-banij/miniconda3/envs/py36/lib/python3.6/site-packages/ipyparallel/engine/kernel.py", line 53, in finish_metadata
    metadata['status'] = reply_content['status']
TypeError: 'Future' object is not subscriptable
2019-04-19 21:23:49.269 [IPEngineApp] ERROR | Exception in message handler:
Traceback (most recent call last):
  File "/gscratch/home/t-banij/miniconda3/envs/py36/lib/python3.6/site-packages/ipykernel/kernelbase.py", line 267, in dispatch_shell
    yield gen.maybe_future(handler(stream, idents, msg))
  File "/gscratch/home/t-banij/miniconda3/envs/py36/lib/python3.6/site-packages/ipyparallel/engine/kernel.py", line 95, in execute_request
    md = self.finish_metadata(parent, md, reply_content)
  File "/gscratch/home/t-banij/miniconda3/envs/py36/lib/python3.6/site-packages/ipyparallel/engine/kernel.py", line 53, in finish_metadata
    metadata['status'] = reply_content['status']
TypeError: 'Future' object is not subscriptable
```

Updating to `ipykernel=4.9.0`  solved the problems.

I am not able to reproduce this on another machine where `ipykernel=5.1.0` is installed.

But it should be reproducible using the following files:
[conda-list.txt](https://github.com/ipython/ipyparallel/files/3099360/conda-list.txt)
[pip-list.txt](https://github.com/ipython/ipyparallel/files/3099361/pip-list.txt)

**Comment by minrk** [comment]
I think this is fixed by #353. I need to make a patch release. It's the IPython version, I believe, not ipykernel that's most relevant.

**Comment by minrk** [comment]
Latest versions are compatible now, including prerelease ipyparallel 7.0a1  with prerelase ipykernel and jupyter-client
